### PR TITLE
Quick Start: Prevent blocked users from scheduling an appointment

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

Check pcgXQF-eV-P2 and 119-gh-concierge-admin.

In this PR:
* Users blocked/banned from Happychat are prevented from purchasing a new Quick Start product.
* Quick Start upsell messages in /me/purchases and /help are not shown for blocked users.
* Users blocked/banned from Happychat with an existing Quick Start product will not be allowed to schedule an appointment.

#### Testing instructions

**Scenario 1 - Prevent blocked users from scheduling an appointment**

1. Apply D66745-code.
2. Login to an account that has an existing Quick Start product. 
3. Modify backend code line by following the instruction given in the "Testing Instructions" section of D66745-code, to force your logged in account to be blocked in Happychat.
4. Now head to `/me/quickstart` and verify that you are not able to schedule an appointment. You should see a screen like this:

Screenshot 2021-09-16 at 5 19 04 PM

**Scenario 2 - Prevent blocked users from purchasing Quick Start**

1. Apply D66745-code and modify the backend code line by following the instruction given in the "Testing Instructions" section of D66745-code
2. Sign in to an account which does not have a Quick Start product. 
3. Navigate to `/checkout/offer-quickstart-session` and attempt to purchase.
4. You should see an error message as shown below:

Screenshot 2021-09-16 at 5 26 15 PM

**Scenario 3 - Upsell banners to not be shown in Calypso**

1. Apply D66745-code and modify the backend code line by following the instruction given in the "Testing Instructions" section of D66745-code
2. Visit `/me/purchases` and `/me/help` and verify that you don't see any upsell banners for Quick Start.

Related to 119-gh-concierge-admin

### CODE REVIEW

**Review by klimeryk — CHANGES_REQUESTED** [review]
Thank you so much on working on this - so many places to update and it's a huge help that you took this on 🙇 

I've ran into some issues while testing, though:

> Scenario 1

Despite verifying that the API definitely returned `true` for `is_happychat_blocked` and the `client/state/data-layer/wpcom/concierge/initial/from-api.js` correctly "transformed" it, when it reached `me/concierge`, the state was missing the `concierge.isHappychatBlocked` field and as a result, I did not get the expected message:

Screenshot 2021-09-15 at 15 59 25

> Scenario 2

Works as expected 👍 

> Scenario 3

While the upsell disappeared, a new nudge appeared in its place! 😅 

Screenshot 2021-09-15 at 15 51 54

Looks like it's the same problem as above with the state missing the `isHappychatBlocked` property 🤔 
I've tried refreshing many times, double-checking everything, but I definitely get the correct field from the API, it just gets lost somewhere before reaching the state 🤔

File: client/me/concierge/shared/no-available-times.js [hunk 1]
```diff
@@ -12,35 +12,51 @@ class NoAvailableTimes extends Component {
...
-										href="/help/contact"
-										tracksEventName="calypso_concierge_book_contact_us"
-									/>
-								),
-							},
-						}
-					) }
+					{ isUserBlocked &&
+						translate(
+							'We add new sessions daily, so please check back soon for more options. In the meantime, consider attending one of our expert webinars on a wide variety of topics designed to help you build and grow your site. {{externalLink1}}View webinars{{/externalLink1}}.',
```
Comment (a8ci18n) [inline client/me/concierge/shared/no-available-times.js:25]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'If you don’t see a day or time that works for you, please check back soon for more options! In the meantime, consider attending one of our expert webinars on a wide variety of topics designed to help you build and grow your site. {{externalLink}}View webinars{{/externalLink}}.' )` *ES Score: 6*

**Comment by niranjan-uma-shankar** [comment]
Thanks for testing @klimeryk .

> Looks like it's the same problem as above with the state missing the isHappychatBlocked property 

You're right, there was a state issue and I've fixed it. The PR is now ready for review again :)

**Comment by klimeryk** [comment]
Thanks for all the updates, @niranjan-uma-shankar! I'm afraid the state is still causing some troubles:

```
Uncaught Error: When called with an action of type "CONCIERGE_INITIAL_UPDATE", the slice reducer for key "isUserBlocked" returned undefined. To ignore an action, you must explicitly return the previous state. If you want this reducer to hold no value, you can return null instead of undefined.
```

Unfortunately, it errors out and gets stuck in loading state :( Maybe try without state rehydration (http://calypso.localhost:3000/me/purchases?flags=force-sympathy) if your state has that key 🤔 I'd love to help more, but my Redux foo is definitely nowhere near your level, so it's probably best if you take a look at it yourself 🙇‍♂️

**Comment by niranjan-uma-shankar** [comment]
> Unfortunately, it errors out and gets stuck in loading state 

That's strange. Could you please confirm if you have D66745-code applied and that your hosts file is pointing to sandbox? The console error and stuck at loading state will happen if your hosts file is pointing to production, but shouldn't happen when pointed to sandbox. I can dig into this again after your confirmation.

**Review by klimeryk — APPROVED** [review]
> That's strange. Could you please confirm if you have D66745-code applied and that your hosts file is pointing to sandbox?

Argh, that was indeed the issue! Sorry for the noise, I could have sworn I applied the patch (but maybe I forgot to sandbox public-api?).

Either way, I've retested and now it works perfectly 💖 (tested both blocked and not blocked cases)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6638024

Thank you @niranjan-uma-shankar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6638024).

**Pull request merged**